### PR TITLE
add support for withHelper for config field

### PR DESCRIPTION
**Pull request by skotambkar**
Updates Config field to support withHelper option it partially supports. The withHelper option is used to generate a With-Helper function for a Config Field.

### CODE REVIEW

File: codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/ConfigField.java [hunk 1]
```diff
@@ -85,12 +85,13 @@ public boolean equals(Object o) {
...
                 && Objects.equals(getType(), that.getType())
-                && Objects.equals(getDocumentation(), that.getDocumentation());
+                && Objects.equals(getDocumentation(), that.getDocumentation())
+                && Objects.equals(withHelper(), that.withHelper());
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(getName(), getType(), getDocumentation());
+        return Objects.hash(getName(), getType(), getDocumentation(), withHelper());
```
[the remark this answers is not in the document]
Comment (skotambkar) [reply codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/ConfigField.java:94]: I think with is self sufficient here, adding get seems redundant.

[the remark this answers is not in the document]
Comment (skmcgrail) [reply codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/ConfigField.java:94]: The pattern for retrieving private fields in java is to prefix with `get<FieldName>`.

[the remark this answers is not in the document]
Comment (jasdel) [reply codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/ConfigField.java:94]: I'm fine with changing it to `get` prefix, When this was original added, I wasn't following the Java naming, and used the implicit behavior.

[the remark this answers is not in the document]
Comment (skotambkar) [reply codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/ConfigField.java]: Alright, updated!

**Review by skmcgrail — APPROVED** [review]

**Review by jasdel — APPROVED** [review]

**Pull request merged**